src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,406 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 -  moving mean square kernel device helpers --------------------
+
+__host__ __device__ int smem_pos(int pos)
+{
+    return pos + (pos >> 5);
+}
+
+__device__ float square(float value)
+{
+    return (value * value);
+}
+
+__device__ void PrefixSum(float *input, uint bufferLength)
+{
+    int offset = 1;
+    int tid = hipThreadIdx_x;
+
+    for (int d = bufferLength >> 1; d > 0; d >>= 1)
+    {
+        __syncthreads();
+        for (int idx = tid; idx < d; idx += hipBlockDim_x)
+        {
+            int ai = offset * (2 * idx + 1) - 1;
+            int bi = offset * (2 * idx + 2) - 1;
+            input[smem_pos(bi)] += input[smem_pos(ai)];
+        }
+        offset <<= 1;
+    }
+
+    if (tid == 0)
+    {
+        int last = bufferLength - 1;
+        input[smem_pos(last)] = 0;
+    }
+
+    for (int d = 1; d < bufferLength; d <<= 1)
+    {
+        offset >>= 1;
+        __syncthreads();
+        for (int idx = tid; idx < d; idx += hipBlockDim_x)
+        {
+            int smem_pos_ai = smem_pos(offset * (2 * idx + 1) - 1);
+            int smem_pos_bi = smem_pos(offset * (2 * idx + 2) - 1);
```

Comment:
Camel case for variables.
And needs more comments in general..

Reply:
changed to camel case